yt/yt/server/controller_agent/controllers/operation_controller_detail.cpp

```diff
@@ -10035,10 +10035,27 @@ void TOperationControllerBase::InitUserJobSpecTemplate(
     jobSpec->set_rpc_proxy_worker_thread_pool_size(jobSpecConfig->RpcProxyWorkerThreadPoolSize);
 
     // Pass external docker image into job spec as is.
-    if (jobSpecConfig->DockerImage &&
-        !TDockerImageSpec(*jobSpecConfig->DockerImage, Config->DockerRegistry).IsInternal())
-    {
-        jobSpec->set_docker_image(*jobSpecConfig->DockerImage);
+    if (jobSpecConfig->DockerImage) {
+        TDockerImageSpec dockerImageSpec(*jobSpecConfig->DockerImage, Config->DockerRegistry);
+        if (dockerImageSpec.IsInternal() && Config->DockerRegistry->UseYtTokenForInternalRegistry) {
+            auto findEnv = [&] (const char* key) -> const TString* {
+                for (auto* environment : {&jobSpecConfig->Environment, &Config->Environment}) {
+                    auto found = environment->find(key);
+                    if (found != environment->end()) {
+                        return &found->second;
+                    }
+                }
+                return nullptr;
+            };
+            if (!findEnv("YT_SECURE_VAULT_docker_auth")) {
```

Comment:
I also vote strongly for making the name of the env variable a shared constant between CA and exec node (`job.cpp`). You can put it somewhere in `server/lib`, like `yt/yt/server/lib/controller_agent/helpers.h`, or just any reasonable place.

Reply:
Yep, add constant for names and convention should be in helper function.